### ISSUE TITLE
Add a `background-type` paint property

### ISSUE THREAD

**Issue by tristen** [mapbox/mapbox-gl-js#11489]
## Motivation

We've heard accounts of it being confusing that a checkered background was shown in Studio when choosing a projection other than `mercator`:

![background](https://user-images.githubusercontent.com/61150/153070617-488bbeeb-0472-419b-8c72-22cf7b2691f9.gif)

We create that checkered background in Studio **but more broadly**: It's slightly weird (exacerbated by `globe`) that a `background` layer wraps around a projection itself and isn't strictly the background styling of the canvas. 

## Proposal

An enum that let's us change the behavior of background to render in governance to its projection or the viewport.

```json
"background-type": {
  "type": "enum",
  "default": "map",
  "values": {
    "map": {
      "doc": "Renders the background layer to the projected map"
    },
    "viewport": {
      "doc": "Renders the background layer to the canvas"
    }
  }
}
```

Doing something would also allow us to apply a `background-pattern` (i.e creating a starry space backdrop).

---

**Mapbox GL JS version:** `2.7.0-alpha.1`

**Comment by asheemmamoowala** [comment]
@tristen How ae you painting the checkered viewport background? Is it possible for a developer to apply this for any projection for the non-map portion of the screen?

**Comment by tristen** [comment]
@asheemmamoowala no. That background is a repeatable pattern we use in Studio (applied to the canvas container in CSS)

**Comment by willymaps** [comment]
Adding that this could also help Mercator map projections as the sky layer defaults as a background color at low-pitched zooms:

![image](https://user-images.githubusercontent.com/12721145/156015214-3c02463a-ff4e-41dc-b882-aa3890b20464.png)
